### PR TITLE
feat(wiki.nixos.org): init

### LINKED ISSUE #781: nixos.wiki is now wiki.nixos.org
### Is there an existing issue outlining your problem?

- [X] I am using the latest version of the provided userstyle.
- [X] I have searched the existing issues and they do not solve my problem.

### What userstyles are you seeing the problem on?

lbl:nixos.wiki

### Describe your problem.

https://nixos.wiki is being replaced by https://wiki.nixos.org, both of these websites use mediawiki but somewhat different css.

### Attach screenshots.

_No response_

### What browser(s) are you seeing the problem on?

N/A

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by orxngc**
## 🎉 Theme for wiki.nixos.org 🎉

The official NixOS Linux distribution wiki. 

## 💬 Additional Comments 💬
I am replacing the old and no longer maintained nixos.wiki (the unofficial NixOS wiki) port with this.

Closes #781.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

**Comment by orxngc** [comment]
It's just the template for now, the code I've already written isn't here yet.. I'll commit it tomorrow, I'm really sleepy.

**Comment by orxngc** [comment]
I think it's mostly complete. Definitely needs more testing though.

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 1]
```diff
@@ -15,17 +15,31 @@
...
+  */
   @media (prefers-color-scheme: light) {
     :root {
       #catppuccin(@lightFlavor, @accentColor);
+      --home-panel-heading-background: @mantle;
+      --home-panel-border-color: @crust;
+      --table-border-color: @crust;
+      --table-header-background: @surface1;
+      --table-even-background: @surface2;
+      --background-color-base: @base;
```
Comment (isabelroses) [inline styles/wiki.nixos.org/catppuccin.user.css:31]: if these need to be in `:root` you can put them in the `#catppuccin()` mixin without any additional selectors

**Review by isabelroses — COMMENTED** [review]
you seem to have some small linter errors.

File: styles/wiki.nixos.org/catppuccin.user.css (hunk 1 above)
Comment (orxngc) [reply styles/wiki.nixos.org/catppuccin.user.css]: Done

**Comment by orxngc** [comment]
> you seem to have some small linter errors.

What do you mean by linter errors?

**Comment by isabelroses** [comment]
> > you seem to have some small linter errors.
> 
> What do you mean by linter errors?

We have a ci that runs `deno task lint`. The errors will show up [in the files view](https://github.com/catppuccin/userstyles/pull/1199/files) you can also automatically fix some of these with `deno task lint:fix`.

**Comment by orxngc** [comment]
Ah, I see it now. I'll fix this when I get back home.

**Comment by orxngc** [comment]
Those linting errors should be fixed.

File: scripts/userstyles.yml [hunk 2]
```diff
@@ -638,14 +641,14 @@ userstyles:
     readme:
       app-link: "https://search.nixos.org"
     current-maintainers: [*cecelot]
-  nixos.wiki:
+  wiki.nixos.org:
     name: NixOS Wiki
     categories: [wiki, productivity]
```
Comment (uncenter) [inline scripts/userstyles.yml:646]:
```suggestion
    categories: [wiki, development]
```

Seems more apt even for the previous wiki :)

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 3]
```diff
@@ -15,9 +15,11 @@
...
 ==/UserStyle== */
 
-@-moz-document domain('nixos.wiki') {
-  @import url("https://python.catppuccin.com/pygments/catppuccin-variables.important.css");
-
+@-moz-document domain('wiki.nixos.org') {
+  /* This section applies the light/dark themes automatically based on the user's system theme.
+     Use the below example CSS to start if the website you are theming doesn't have a light/dark mode toggle.
+     For an actual example of this, see https://github.com/catppuccin/userstyles/blob/39288b593278e5efa48007d5fc001292cf910672/styles/brave-search/catppuccin.user.css#L16-L29.
+  */
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:22]: No need to keep the tutorial comments after filling it out
```suggestion
```

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 4]
```diff
@@ -57,33 +59,12 @@
...
-    --ctp-subtext0: @subtext0;
-    --ctp-overlay2: @overlay2;
-    --ctp-overlay1: @overlay1;
-    --ctp-overlay0: @overlay0;
-    --ctp-surface2: @surface2;
-    --ctp-surface1: @surface1;
-    --ctp-surface0: @surface0;
-    --ctp-base: @base;
-    --ctp-mantle: @mantle;
-    --ctp-crust: @crust;
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:86]: These variables are key to adding syntax highlighting, see https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments.

**Review by uncenter — CHANGES_REQUESTED** [review]

**Comment by orxngc** [comment]
@uncenter Resolved all requested changes.

**Comment by uncenter** [comment]
Seeing a lot of unthemed elements: borders/dividers, icons. The icons may be tricky, https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#how-do-i-theme-images-and-svgs should help.
CleanShot 2024-08-17 at 07 35 45@2x

**Comment by orxngc** [comment]
Will do these soon.. ty for pointing them out!

**Comment by orxngc** [comment]
Wh—
Did I really forget about this for four days? Sheesh
Let's see if I have le time to do this now 😅

**Comment by orxngc** [comment]
I.. think did all of that.

**Comment by isabelroses** [comment]
Currently all the borders I can see are too dark i.e. crust or seem to be unthemed.

The history pages seems to be unthemed. [example url](https://wiki.nixos.org/w/index.php?title=NixOS_Wiki&action=history)
image

The logs page too. [example url](https://wiki.nixos.org/w/index.php?title=Special:Log&page=NixOS+Wiki)
image

**Comment by orxngc** [comment]
Ah.. thanks, I'll get to these.

**Comment by orxngc** [comment]
I redid the whole thing by using the Wikipedia port as a base. The wikipedia maintainer writes much cleaner code than I do..

**Comment by orxngc** [comment]
Shoot, latte is broken, one second.

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 5]
```diff
@@ -0,0 +1,1466 @@
...
+    --ctp-overlay0: @overlay0;
+    --ctp-surface2: @surface2;
+    --ctp-surface1: @surface1;
+    --ctp-surface0: @surface0;
+    --ctp-base: @base;
+    --ctp-mantle: @mantle;
+    --ctp-crust: @crust;
+    --background-color-base: @base;
+    --home-panel-heading-background: @mantle;
+    --home-panel-border-color: @crust;
```
Comment (isabelroses) [inline styles/wiki.nixos.org/catppuccin.user.css:89]: could we make this a lighter color like `@surface0` so its clearer

**Review by isabelroses — COMMENTED** [review]
most other things seem good if they were based on wikipedia port

File: styles/wiki.nixos.org/catppuccin.user.css (hunk 5 above)
Comment (orxngc) [reply styles/wiki.nixos.org/catppuccin.user.css:89]: Which one of these do you mean when you refer to "this"? ALl of them?

Comment (orxngc) [reply styles/wiki.nixos.org/catppuccin.user.css:89]: @isabelroses

Comment (isabelroses) [reply styles/wiki.nixos.org/catppuccin.user.css:89]: sorry for the late reply but it's what uncenter said, I didn't realise GitHub would make this so unclear.

Comment (orxngc) [reply styles/wiki.nixos.org/catppuccin.user.css]: Nw, got it thanks

**Review by isabelroses — COMMENTED** [review]
Only thing I can seem to spot is this. It would be nice if we used crust here for readablity.
image

**Comment by orxngc** [comment]
Gimme a moment then

**Review by isabelroses — APPROVED** [review]

**Comment by orxngc** [comment]
Oh checks are failing ic.. gimme a second whoopsies

**Comment by orxngc** [comment]
image
Okay
image
Interesting

**Comment by uncenter** [comment]
You are missing the `supportURL` field: https://github.com/catppuccin/userstyles/blob/9cc2b42a3d8722a5b599c1b3735383a26dc6e2ac/template/catppuccin.user.css#L6.

**Comment by orxngc** [comment]
Ah I see

**Comment by orxngc** [comment]
Fixed

**Review by uncenter — CHANGES_REQUESTED** [review]
On the home page of the wiki, still seeing a few unthemed things.
CleanShot 2024-08-23 at 16 37 22@2x

The top border here, and the blue dots next to the language options.

**Comment by orxngc** [comment]
Thanks, I got those now.

**Comment by orxngc** [comment]
Oh no I didn't, the codeblocks! One second

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 6]
```diff
@@ -0,0 +1,1480 @@
...
+    }
+    .oo-ui-menuOptionWidget:hover {
+      background-color: @surface1;
+      color: @text;
+    }
+    .cdx-search-input__end-button {
+      background-color: @crust !important;
+    }
+    .mw-pt-languages {
+      border-bottom: 1px solid @surface0;
```
Comment (isabelroses) [inline styles/wiki.nixos.org/catppuccin.user.css:1459]:
```suggestion
      border-bottom-color: @surface0;
```

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 7]
```diff
@@ -0,0 +1,1480 @@
...
+    }
+    .mw-pt-progress--complete::after,
+    .mw-pt-progress--high::after {
+      border: 1px solid @accent-color;
```
Comment (isabelroses) [inline styles/wiki.nixos.org/catppuccin.user.css:1463]:
```suggestion
      border-color: @accent-color;
```

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 8]
```diff
@@ -0,0 +1,1480 @@
...
+      background: conic-gradient(@text 0, @accent-color 0);
+    }
+    pre {
+      border: 1px solid @surface0;
```
Comment (isabelroses) [inline styles/wiki.nixos.org/catppuccin.user.css:1467]:
```suggestion
      border-color: @surface0;
```

**Review by isabelroses — CHANGES_REQUESTED** [review]

**Comment by isabelroses** [comment]
Hey, seems we have a merge conflict, can you fix that.

**Comment by isabelroses** [comment]
> Only thing I can seem to spot is this. It would be nice if we used crust here for readablity. image

I think we regressed on this, still seems broken.

**Comment by orxngc** [comment]
> @isabelroses I think we regressed on this, still seems broken.

Are you using the version as of the latest commit? I fixed this issue, it displays as crust for me.

**Comment by orxngc** [comment]
image

**Review by isabelroses — DISMISSED** [review]

**Comment by isabelroses** [comment]
> > I think we regressed on this, still seems broken.
> 
> @isabelroses Are you using the version as of the latest commit? I fixed this issue, it displays as crust for me.

Sorry... I had 3 versions installed. Everything seems good now.

**Comment by orxngc** [comment]
Checks do B failing again.. let's see

**Comment by orxngc** [comment]
Yupp.. there we go, fixed O.O

**Review by isabelroses — APPROVED** [review]

**Comment by somerand0mcat** [comment]
nice work! all you have to do is to fix these unthemed icons on the home page:
image
image
image

**Comment by uncenter** [comment]
Omar I've chosen not to do that since that is, again, undocumented and difficult to do. Please stop recommending it when you are not userstyles staff nor are you even able to view userstyles updates and discussions in the Discord.

**Comment by somerand0mcat** [comment]
> Omar I've chosen not to do that since that is, again, undocumented and difficult to do. Please stop recommending it when you are not userstyles staff nor are you even able to view userstyles updates and discussions in the Discord.

fine, im sorry

**Comment by orxngc** [comment]
@uncenter Is there anything else that needs to be done here..? >,>

**Comment by isabelroses** [comment]
> @uncenter Is there anything else that needs to be done here..? >,>

They mentioned they were on break, so just be patient for the next few days and they will get back to you 🙏

**Comment by orxngc** [comment]
My bad— alright! 💖

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 9]
```diff
@@ -0,0 +1,1480 @@
...
+    }
+
+    .vector-feature-zebra-design-enabled .vector-pinnable-header,
+    .vector-pinnable-header,
+    .vector-pinnable-element .vector-menu-heading {
+      border-bottom-color: @surface0;
+    }
+
+    hr {
+      border-bottom-color: @surface2;
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:814]: I believe this should be background color (or at least thats how the one on https://wiki.nixos.org/w/index.php?title=NixOS_Wiki&action=edit is themed).

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 10]
```diff
@@ -0,0 +1,1480 @@
...
+      background-color: @surface1 !important;
+      border-color: @surface2 !important;
+    }
+
+    .navbox-title {
+      color: @text !important;
+    }
+
+    .mw-highlight pre {
+      background: @surface0 !important;
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:365]:
```suggestion
      background: @mantle !important;
```

I think I prefer the contrast of this.

**Review by uncenter — CHANGES_REQUESTED** [review]
The border around boxes like "Filter revisions" on https://wiki.nixos.org/w/index.php?title=NixOS_Wiki&action=history or https://wiki.nixos.org/wiki/Special:WhatLinksHere/NixOS_Wiki is not themed.

The three dots/menu button in the top left on any page (scroll down a bit) is unthemed:
CleanShot 2024-09-02 at 18 47 14@2x

**Comment by orxngc** [comment]
> The three dots/menu button in the top left on any page (scroll down a bit) is unthemed:

I cannot find this button lol

File: styles/wiki.nixos.org/catppuccin.user.css (hunk 9 above)
Comment (uncenter) [reply styles/wiki.nixos.org/catppuccin.user.css]: Still is border-bottom-color but now with a different color, I believe it should be `background-color`.

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 11]
```diff
@@ -0,0 +1,1480 @@
...
+
+    .mw-highlight pre {
+      background: @mantle !important;
+      border-width: 0;
+      color: @text;
+    }
+
+    .mw-content-ltr.mw-highlight-lines pre,
+    .mw-content-ltr.content .mw-highlight-lines pre {
+      box-shadow: inset 2.75em 0 0 @surface1;
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:372]:
```suggestion
      box-shadow: inset 2.75em 0 0 @mantle;
```

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 12]
```diff
@@ -0,0 +1,1480 @@
...
+    .navbox-title {
+      color: @text !important;
+    }
+
+    .mw-highlight pre {
+      background: @mantle !important;
+      border-width: 0;
+      color: @text;
+    }
+
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:369]:
```suggestion
```
Pretty sure all of this is already themed without this section.

**Review by uncenter — CHANGES_REQUESTED** [review]
The "The NixOS core ecosystem stack" diagram/table on https://wiki.nixos.org/wiki/Nix_ecosystem#Official_ecosystem is unthemed.

The info alert and TLDR box on https://wiki.nixos.org/wiki/Nix_Cookbook are unthemed.

**Comment by orxngc** [comment]
> The "The NixOS core ecosystem stack" diagram/table on https://wiki.nixos.org/wiki/Nix_ecosystem#Official_ecosystem is unthemed.
> 
> The info alert and TLDR box at the top of https://wiki.nixos.org/wiki/Nix_Cookbook are unthemed.

will deal with them tomorrow

**Comment by orxngc** [comment]
> The info alert and TLDR box at the top of https://wiki.nixos.org/wiki/Nix_Cookbook are unthemed.

I honestly have no idea how to select these elements, they're divs with inline style attributes
they don't have any class or ID



> The "The NixOS core ecosystem stack" diagram/table on https://wiki.nixos.org/wiki/Nix_ecosystem#Official_ecosystem is unthemed.

done



> The three dots/menu button in the top left on any page (scroll down a bit) is unthemed:

also done

> Let's make sure to retake this now that most of that page is now properly themed, and let's make sure we use a regular laptop resolution for the images (looks like you maybe got a full page screenshot?).

yes i was taking full page screenshots, i'll take normal ones then yep

**Comment by uncenter** [comment]
For inline style elements you can do:
```css
/* <Description of where inline styles can be found> */
[style*="background-color: #C1E5FF;"] {
  background-color: @whatever;
}
[style*="border: thin solid #1D99F3;"] {
  border-color: @whatever;
}
```

**Comment by orxngc** [comment]
Err.. seems like I forgot about this for two weeks.. my bad kek

File: scripts/userstyles.yml [hunk 13]
```diff
@@ -993,4 +996,4 @@ userstyles:
           answer: "It's available in case you have an OLED display. If you have one, you might want to enable this."
     current-maintainers: [*isabelroses, *uncenter]
     past-maintainers: [*elkrien]
-# yaml-language-server: $schema=userstyles.schema.json
+# yaml-language-server: $schema=userstyles.schema.json
```
Comment (uncenter) [inline scripts/userstyles.yml:999]: Please add this trailing newline back :)

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 14]
```diff
@@ -0,0 +1,1492 @@
...
+    @macchiato: { @rosewater: #f4dbd6; @flamingo: #f0c6c6; @pink: #f5bde6; @mauve: #c6a0f6; @red: #ed8796; @maroon: #ee99a0; @peach: #f5a97f; @yellow: #eed49f; @green: #a6da95; @teal: #8bd5ca; @sky: #91d7e3; @sapphire: #7dc4e4; @blue: #8aadf4; @lavender: #b7bdf8; @text: #cad3f5; @subtext1: #b8c0e0; @subtext0: #a5adcb; @overlay2: #939ab7; @overlay1: #8087a2; @overlay0: #6e738d; @surface2: #5b6078; @surface1: #494d64; @surface0: #363a4f; @base: #24273a; @mantle: #1e2030; @crust: #181926; };
+    @mocha:     { @rosewater: #f5e0dc; @flamingo: #f2cdcd; @pink: #f5c2e7; @mauve: #cba6f7; @red: #f38ba8; @maroon: #eba0ac; @peach: #fab387; @yellow: #f9e2af; @green: #a6e3a1; @teal: #94e2d5; @sky: #89dceb; @sapphire: #74c7ec; @blue: #89b4fa; @lavender: #b4befe; @text: #cdd6f4; @subtext1: #bac2de; @subtext0: #a6adc8; @overlay2: #9399b2; @overlay1: #7f849c; @overlay0: #6c7086; @surface2: #585b70; @surface1: #45475a; @surface0: #313244; @base: #1e1e2e; @mantle: #181825; @crust: #11111b; };
+  }
+
+// vim:ft=less
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:1492]: Trailing newlines please! Your editor *should* be configured to support https://editorconfig.org/ if possible, which would respect this setting.

File: styles/wiki.nixos.org/catppuccin.user.css [hunk 15]
```diff
@@ -0,0 +1,1492 @@
...
+    --ctp-base: @base;
+    --ctp-mantle: @mantle;
+    --ctp-crust: @crust;
+    --background-color-base: @base;
+    --home-panel-heading-background: @mantle;
+    --home-panel-border-color: @surface0;
+    --table-border-color: @crust;
+    --table-header-background: @surface1;
+    --table-even-background: @surface2;
+    background-color: @base;
```
Comment (uncenter) [inline styles/wiki.nixos.org/catppuccin.user.css:94]:
```suggestion
    --ctp-crust: @crust;

    --background-color-base: @base;
    --home-panel-heading-background: @mantle;
    --home-panel-border-color: @surface0;
    --table-border-color: @crust;
    --table-header-background: @surface1;
    --table-even-background: @surface2;

    background-color: @base;
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Comment by uncenter** [comment]
Please avoid unnecessarily updating this branch as it pings people who are subscribed to it :)

File: scripts/userstyles.yml [hunk 16]
```diff
@@ -664,14 +667,14 @@ userstyles:
...
     current-maintainers: [*cecelot]
-  nixos.wiki:
+  wiki.nixos.org:
     name: NixOS Wiki
-    categories: [wiki, productivity]
+    categories: [wiki, development]
     icon: nixos
     color: blue
     readme:
       app-link: "https://nixos.wiki"
```
Comment (uncenter) [inline scripts/userstyles.yml:676]:
```suggestion
      app-link: "wiki.nixos.org"
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by uncenter — APPROVED** [review]

**Comment by uncenter** [comment]
Approving for the short term, really hoping at some point we can get MediaWiki updated on wiki.nixos.org and just use #1075 for it 😭

**Pull request merged**